larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/ShowerESTAREnergy_tool.cc

```diff
@@ -0,0 +1,293 @@
+      if(showercentre.Mag() >= 0 && chargecentre.Mag() >=0){ 
+        localEfield = IShowerTool::GetLArPandoraShowerAlg().SCECorrectEField(nominal_Efield, showercentre);
+        localEfield_cweighted = IShowerTool::GetLArPandoraShowerAlg().SCECorrectEField(nominal_Efield, chargecentre);
+      }
+      else{
+        mf::LogWarning("ShowerESTAREnergy") << "Shower centre calculation doesn't look to be sensible. Reconstruction is probably dodgy." << std::endl;
+      }
+    }
+
+    // Loop over the clusters in the plane and get the hits
+    for(auto const& cluster: clusters){
+          
+      // Get the hits
+      std::vector<art::Ptr<recob::Hit> > hits = fmhc.at(cluster.key());
+      // Get the plane.
+      const geo::PlaneID::PlaneID_t plane(cluster->Plane().Plane);
+       
+      planeHits[plane].insert(planeHits[plane].end(),hits.begin(),hits.end());
+    }
+
+    // Loop over the spacepoints and get the hits
+    sp_hits_tuple.clear();
+    for(auto const& spacepoint : spacepoints){
+      //Get the hits
+      std::vector<art::Ptr<recob::Hit> > hits = fmhsp.at(spacepoint.key());
+
+      // get local Efield at SP_pos
+      TVector3 sp_pos = IShowerTool::GetLArPandoraShowerAlg().SpacePointPosition(spacepoint); 
+      double localEfield_SP = IShowerTool::GetLArPandoraShowerAlg().SCECorrectEField(nominal_Efield, sp_pos);
+
+      // Store the localEfield of all the hits with associated SP's so we can compare with the hits from the clusters later
+      for(auto const& h : hits){
+        sp_hits_tuple.push_back(std::make_tuple(h.key(), localEfield_SP));
+      }           
+    }
+
+    // Calculate the energy for each plane && best plane
+    geo::PlaneID::PlaneID_t bestPlane  = std::numeric_limits<geo::PlaneID::PlaneID_t>::max();
+    unsigned int bestPlaneNumHits = 0;
+
+    //Holder for the final product
+    std::vector<double> ShowerESTAREnergy(fGeom->Nplanes(), -999);
+    std::vector<double> ShowerESTAREnergyError(fGeom->Nplanes(), -999);
+
+    for(auto const& [plane, hits] : planeHits){
+
+      unsigned int planeNumHits = hits.size();
+        
+      //Calculate the Energy
+      double Energy = CalculateEnergy(clockData, detProp, hits, plane);
+
+      if(Energy > 0){    
+        ShowerESTAREnergy.at(plane) = Energy;
+      }
+
+      if(planeNumHits > bestPlaneNumHits){
+        bestPlane        = plane;
+        bestPlaneNumHits = planeNumHits;
+      }
+      if(fVerbose){
+        std::cout << "plane: " << plane
+                  << "  hits: " << hits.size()
+                  << "  energy: " << Energy << std::endl;  
+      }
+
+    }
+
+    ShowerEleHolder.SetElement(ShowerESTAREnergy,ShowerESTAREnergyError,fShowerEnergyOutputLabel);
+
+    // Only set the best plane if it has some hits in it
+    if (bestPlane < fGeom->Nplanes()){
+      // Need to cast as an int for legacy default of -999
+      // have to define a new variable as we pass-by-reference when filling
+      int bestPlaneVal(bestPlane);
```

Comment:
const?

Reply:
I see you've initialised this var, but elsewhere in the codebase you assign.  Initialising is more efficient buts its probably better to be consistent everywhere...

Reply:
In this instance I would go in the other direction and convert everything else, preferably with curly brace notation as this prevents narrowing unless you explicitly allow it. Initialisation with simple parentheses is definitely ok for object constructor-style calls, as that's more in keeping with calling a function.